va/validation-authority.go

```diff
@@ -333,6 +333,15 @@ func (va *ValidationAuthorityImpl) validateSimpleHTTP(identifier core.AcmeIdenti
 		return challenge, err
 	}
 
+	if httpResponse.Header.Get("Content-Type") != "application/jose+json" {
```

Comment:
I'd like to also detect and reject the case where there are multiple, conflicting values for this header. Per the docs, sounds like you'll have to use CanonicalHeaderKey.